### PR TITLE
Build and test fixes

### PR DESCRIPTION
**Pull request by baronfel**
This PR pull out parts of the multipart form data PR that fix the following issues:

Broken Tests on travis (typically due to casing or yahoo's stock API being deprecated)
Adding linux support to the build (mostly requires package installs and path locations in the test helper modules)
Package updates (including the mono 5 type provider fix)
Paket-referencing FSharp.Core (which helps solve at least two of the above issues :))

### CODE REVIEW

**Comment by baronfel** [comment]
Handles #1054 and #1051, at a minimum.

**Comment by baronfel** [comment]
anyway, @ovatsus linux support is super close. There's some discrepancy between the search.atom.xml file on the file system and in the tests, and the linux runners for the current HEAD of this branch are having errors starting up, but the OSX builds are good 👍

**Comment by dsyme** [comment]
@baronfel Fantastic work! :)

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,38 @@
 language: csharp
 
+sudo: required
+dist: trusty
+
+mono:
+  - 4.4.2
```
Comment (dsyme) [inline .travis.yml]: Can you add "latest" and even "nightly" here too?  Perhaps drop one or two of the others

**Comment by baronfel** [comment]
holy smokes all the casing issues are fixed up now! we're successfully building and testing on linux!

**Comment by baronfel** [comment]
as far as I am concerned this PR is complete.

**Comment by ovatsus** [comment]
Thanks, I'll try to review and merge later today

**Comment by ovatsus** [comment]
Since we're now referencing FSharp.Core from NuGet, shouldn't we also update the nuspec so consumers of FSharp.Data also consume FSharp.Core from NuGet, otherwise they can have mismatches, no?

**Comment by baronfel** [comment]
That's a good question.  In the previous version, we built and deployed against FSharp.Core version 4.3.0.0, and any users on further-ahead versions had to do binding redirects.  I think if anything, I should update the nuspec to show that we have an FSharp.Core dependency and let clients at least have that truth, instead of not directly exposing our dependency on FSharp.Core.

**Comment by baronfel** [comment]
And now that I read what you said on my laptop and not my phone, I realize that's exactly what you said.

**Comment by baronfel** [comment]
can you sanity check that nuspec change for me? I haven't found a way to extract the nuget package on OSX yet to verify the contents

**Comment by ovatsus** [comment]
CI was green on the PR, but after merging the CI is failing on Mono 5.01 on Mac

**Comment by baronfel** [comment]
I'll take a look

**Comment by baronfel** [comment]
Seems to be the same intermittent DNS failure that plagued a few of the
Linux builds on the PR. Can you retrigger the build?

**Comment by ovatsus** [comment]
Yep, that was it, after retrying it passed

**Pull request merged**